dotnet/src/SemanticKernel.Skills/Skills.OpenAPI/OpenApi/OpenApiDocumentParser.cs

```diff
@@ -21,21 +26,88 @@ namespace Microsoft.SemanticKernel.Skills.OpenAPI.OpenApi;
 internal class OpenApiDocumentParser : IOpenApiDocumentParser
 {
     /// <inheritdoc/>
-    public IList<RestApiOperation> Parse(Stream stream)
+    public async Task<IList<RestApiOperation>> ParseAsync(Stream stream, CancellationToken cancellationToken = default)
     {
-        var openApiDocument = new OpenApiStreamReader().Read(stream, out var diagnostic);
+        var jsonObject = await this.DowngradeDocumentVersionToSuportedOneAsync(stream, cancellationToken).ConfigureAwait(false);
```

Comment:
Could you also update the title of the PR to replace "Support" with "Mitigate" since we don't *technically* support true OpenAPI 3.1. Of course, let me know if I'm misunderstanding.